### ISSUE TITLE
Publish test output as JUnit XML report

### ISSUE THREAD

**Issue by jeremyhiggs** [solenopsis/Solenopsis#220]
It would be great if Solenopsis published the test results in the JUnit XML format, so that continuous integration software (Jenkins, in my case) can parse and publish the test results.

I looked around for this, and thought the sf.generateTests property would implement this, but it seems not!

**Comment by pcon** [comment]
I don't think this is something (retrieving test results) that the ant interface supports.  I know that the Metadata API does support this, so we'd have to start making use of that, and that's not territory we've gone into.  As a stop-gap, I've used [jsforce](https://jsforce.github.io/) with node to get test coverage. I'm sure you could do something similar to get the results of the test run and generate the JUnit XML needed.  If you want to take a look over the [getTestResults.js](https://github.com/apex-lodash/lo/blob/master/build/scripts/testing/getTestResults.js) and [run_test.sh](https://github.com/apex-lodash/lo/blob/master/build/scripts/run_tests.sh) that I've used with [TravisCI](https://travis-ci.org/), they might give you a start.  The script pushes the results to [Coveralls](coveralls.io), but I'm sure you could modify it easily enough to instead output the JUnit XML.

**Comment by jeremyhiggs** [comment]
Thanks for the tips! I'll take a look at the scripts that you posted and see if I can modify those.

**Comment by sfloess** [comment]
@jeremyhiggs / @pcon,

The Metadata API is something high on my priority list...  I'm going to basically replace Lasius with whats in Keraiai and gut what's there now.  Metadata is something I need pretty badly - the final stages of what I need to do is being formalized now (working one last issue in Keraiai)...
